Made EarlyDeleteHelper thread-safe

Dr15Jones opened the pull request:
Switched to using atomic counters in order to avoid race conditions once we start using more than one thread per event.

Dr15Jones committed:
Made EarlyDeleteHelper thread-safe

Switched to using atomic counters in order to avoid race conditions once we start using more than one thread per event.

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for CMSSW_8_1_X.

It involves the following packages:

FWCore/Framework


@cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @wddgit, @wmtan this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

Dr15Jones committed:
Using meaningful names in BranchToCount

Changed names from first,second to branch,count to allow better self documentation.

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: b3c5e63330e9d87cd8ecea289a80f3011751dd3f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13727/11903/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13727/11903/summary.html

davidlange6 commented:
+1
